lib/ex_aws/s3.ex

```diff
@@ -1283,6 +1283,22 @@ defmodule ExAws.S3 do
 
     credential = ExAws.Auth.Credentials.generate_credential_v4("s3", config, datetime)
 
+    # security_token will be present when temporary credentials are used
+    {opts, security_token_fields} =
+      if config[:security_token] do
+        {
+          Keyword.put(
+            opts,
+            :custom_conditions,
+            Keyword.get(opts, :custom_conditions, []) ++
+              [%{"X-Amz-Security-Token" => config[:security_token]}]
+          ),
+          %{"X-Amz-Security-Token" => config[:security_token]}
```

Comment:
This can be simplified using `Keyword.update`:

```suggestion
          token_map = %{"X-Amz-Security-Token" => config[:security_token]}
          Keyword.update(opts, :custom_conditions, token_map, fn cc -> cc ++ [token_map] end)
```

Reply:
Thanks for that suggestion @hoyon. I've made that change.